docs/source/_auto_api_templates/python/module.rst

```diff
@@ -0,0 +1,120 @@
+{% if not obj.display %}
+:orphan:
+
+{% endif %}
+{% if obj.name.split('.')[-1][0] == '_' %}
+{{ obj.name }}
+{% elif obj.name.split('.')|length <= 2 %}
+{{ (' '.join(obj.name.split('.')[-1].split('_'))|title) }}
```

Comment:
This template is to modify the display names of the modules, e.g. `evalml.model_understanding` to `Model Understanding`